### ISSUE TITLE
Completion fails if function is optional

### ISSUE THREAD

**Issue by haxiomic** [vshaxe/vshaxe#409]
If a function has `@:optional` metadata, completion fails for that type. In the following example, there will be no completion for `example`, but if you remove `@:optional`, completion works.

```haxe
class Main {
    
    static function main() {
        var example: {
            function test(): Void;
            @:optional
            function optionalTest(): Void;
        } = null;

        example.|
    }

}
```

I had a similar bug in my sublime plugin – I assumed if `kind` is 'method', then `type` will be a function signature, however the function signature is wrapped in `Null<T>` so it fails to parse as a function. I.e., the type returned from haxe --display of the `method` `optionalTest()` is `Null<Void -> Void`, which doesn't parse as a function signature

**Comment by haxiomic** [comment]
Seems to be happening again in v2.21.0

**Comment by Gama11** [comment]
Looks like the fix for #449 covers this too.

**Comment by haxiomic** [comment]
Awesome stuff! Thanks @Gama11 ❤️
